sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveCatalog.scala

```diff
@@ -182,13 +189,15 @@ private[spark] class HiveCatalog(client: HiveClient) extends ExternalCatalog wit
     client.getTable(db, table)
   }
 
+  override def tableExists(db: String, table: String): Boolean = withClient {
+    client.getTableOption(db, table).isDefined
+  }
+
   override def listTables(db: String): Seq[String] = withClient {
-    requireDbExists(db)
```

Comment:
seems like we should throw exception when db doesn't exist.

we could do that as a separate pr though